eng/pipelines/templates/steps/build.yml

```diff
@@ -12,6 +12,7 @@ steps:
         pwsh: true
         workingDirectory: $(Build.SourcesDirectory)
         filePath: eng/scripts/SetTestPipelineVersion.ps1
+        arguments: '-BuildNumber $(Build.BuildNumber)'
```

Comment:
The .NET repo used BuildID, was using BuildNumber intentional here?

https://github.com/Azure/azure-sdk-for-net/pull/16854#pullrequestreview-530524625

Reply:
While title says buildid it actually uses BuildNumber which is I believe the correct thing.